libs/smart-ngrx/src/actions/action.service.ts

```diff
@@ -412,5 +412,12 @@ export class ActionService {
         ids: idsToRemove,
       }),
     );
+    // make sure we remove the virtualArray from the map AFTER
+    // we remove the row from the store
+    asapScheduler.schedule(() => {
+      idsToRemove.forEach((id) => {
+        virtualArrayMap.remove(this.feature, this.entity, id);
+      });
+    });
```

Comment:
**suggestion (bug_risk):** Consider potential race conditions with asynchronous virtualArrayMap removal

Using asapScheduler to remove entries from virtualArrayMap after dispatching the remove action could potentially lead to race conditions. Consider whether it's possible to remove the entries synchronously after the action dispatch, or if not, implement additional safeguards to ensure consistency between the store and virtualArrayMap.